Update linker-plugin-lto.md to contain up to Rust 1.65

str4d opened the pull request:
The table rows were obtained via the script embedded in the page.

str4d committed:
Update linker-plugin-lto.md to contain up to Rust 1.65

The table rows were obtained via the script embedded in the page.

str4d commented:
This table is starting to get a bit unwieldy, though there are much larger tables elsewhere in the book, so maybe it's fine. For now I've kept updating it per the script embedded in the page, but we could alternatively switch to a range-based table, if that fits with how Rust does LLVM updates:

| Rust Version     | Clang Version |
|------------------|---------------|
| Rust 1.34 - 1.37 |    Clang 8    |
| Rust 1.38 - 1.44 |    Clang 9    |
| Rust 1.45 - 1.46 |    Clang 10   |
| Rust 1.47 - 1.51 |    Clang 11   |
| Rust 1.52 - 1.55 |    Clang 12   |
| Rust 1.56 - 1.59 |    Clang 13   |
| Rust 1.60 - 1.64 |    Clang 14   |
| Rust 1.65        |    Clang 15   |

ehuss commented:
The range-based table definitely looks a lot nicer. I imagine it might make it a little more awkward if someone just wants to search for a specific version, but I don't think that should be too much of a concern. If you want, would you mind switching it to that?

str4d committed:
Migrate linker-plugin-lto.md compatibility table to show Rust ranges

The helper shell script has been rewritten as a helper Python script
that generates the range-based table.

str4d commented:
Done. I also updated the script to generate the range-based table.

bors commented:
:pushpin: Commit ee7a80211a066b8345119ce258361a8ad22986b4 has been approved by `ehuss`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=ehuss ee7a80211a066b8345119ce258361a8ad22986b4 -->
<!-- homu: {"type":"Approved","sha":"ee7a80211a066b8345119ce258361a8ad22986b4","approver":"ehuss","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
